src/games/strategy/triplea/delegate/Matches.java

```diff
@@ -639,7 +643,8 @@ public boolean match(final Unit obj) {
       public boolean match(final Unit obj) {
         final Unit unit = obj;
         final UnitAttachment ua = UnitAttachment.get(unit.getType());
-        return !ua.getIsInfrastructure()
+        // Ensure unit is "attackable" when they have more than one hit point (even if it is infrastructure)
+        return (!ua.getIsInfrastructure() || ua.getHitPoints() > 1)
```

Comment:
if we have a match for this condition:
`!ua.getIsInfrastructure() || ua.getHitPoints() > 1`

Then we can call that matcher here, and also above https://github.com/triplea-game/triplea/pull/1336/files#diff-974901706eaaa738871895c34df741cbR64  `new CompositeMatchOr<Unit>(Matches.UnitIsNotInfrastructure, Matches.UnitHasMoreThanOneHitPointTotal));`